enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,529 @@
+---
+title: assisted-installer-in-cluster
+authors:
+  - "@mhrivnak"
+  - "@hardys"
+  - "@avishayt"
+reviewers:
+  - "@eparis"
+  - "@markmc"
+  - "@dgoodwin"
+  - "@ronniel1"
+
+approvers:
+  - TBD
+creation-date: 2020-12-22
+last-updated: 2021-01-10
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced|informational
+see-also:
+replaces:
+superseded-by:
+---
+
+# Assisted Installer in-cluster
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The Assisted Installer currently runs as a SaaS on cloud.redhat.com, enabling
+users to deploy OpenShift clusters with certain customizations, particularly on
+bare metal hardware. It is necessary to bring those capabilities on-premise in
+users' clusters for two purposes:
+
+1. A cluster created by the Assisted Installer needs the ability to add workers
+on day 2.
+2. Multi-cluster management, such as through hive and RHACM, should be
+able to utilize the capabilities of the Assisted Installer.
+
+This enhancement proposes the first iteration of running the Assisted Installer
+in end-user clusters to meet the purposes above.
+
+## Motivation
+
+### Goals
+
+* Expose the Assisted Installer's capabilities as a kubernetes-native API. Some
```

Comment:
I'm not sure why we say that this is a goal in this document. I presume it is because we expect RHACM to require it? Is that the only reason?

Reply:
RHACM will be a main user of the API, but not necessarily the only one. For example there are also the stand-alone cluster use cases described below where a cluster just needs to add workers to itself. An end user or other tooling may need to interact with these APIs.

Since OpenShift has adopted a policy of doing infrastructure management through k8s-native APIs, it is seen as a requirement for these capabilities to likewise follow that pattern.

Reply:
Right, I think this was left over from the first draft of this document.  Since then we decided on exposing agent-based via Hive's ClusterDeployment, plus a couple of other supporting CRDs.